src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -0,0 +1,326 @@
+        $ctorArguments = [];
+        $constructor = $targetRefl->getConstructor();
+        foreach ($constructor?->getParameters() ?? [] as $parameter) {
+            $parameterName = $parameter->getName();
+            if (!$targetRefl->hasProperty($parameterName)) {
+                continue;
+            }
+
+            $property = $targetRefl->getProperty($parameterName);
+
+            // The mapped class was probably instantiated in a transform we can't write a readonly property
+            if ($property->isReadOnly() && ($property->isInitialized($mappedTarget) && $property->getValue($mappedTarget))) {
+                continue;
+            }
+
+            $ctorArguments[$parameterName] = $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+        }
+
+        $readMetadataFrom = $source;
+        $refl = $this->getSourceReflectionClass($source, $targetRefl);
+
+        // When source contains no metadata, we read metadata on the target instead
+        if ($refl === $targetRefl) {
+            $readMetadataFrom = $mappedTarget;
+        }
+
+        $mapToProperties = [];
+        foreach ($refl->getProperties() as $property) {
+            if ($property->isStatic()) {
+                continue;
+            }
+
+            $propertyName = $property->getName();
+            $mappings = $this->metadataFactory->create($readMetadataFrom, $propertyName);
+            foreach ($mappings as $mapping) {
+                $sourcePropertyName = $propertyName;
+                if ($mapping->source && (!$refl->hasProperty($propertyName) || !isset($source->$propertyName))) {
+                    $sourcePropertyName = $mapping->source;
+                }
+
+                $value = $this->getRawValue($source, $sourcePropertyName);
+                if (($if = $mapping->if) && ($fn = $this->getCallable($if, $this->conditionCallableLocator)) && !$this->call($fn, $value, $source)) {
+                    continue;
+                }
+
+                if (false === $if) {
+                    continue;
+                }
+
+                $targetPropertyName = $mapping->target ?? $propertyName;
+                if (!$targetRefl->hasProperty($targetPropertyName)) {
+                    continue;
+                }
+
+                $value = $this->getSourceValue($source, $mappedTarget, $value, $objectMap, $mapping);
+                $this->storeValue($targetPropertyName, $mapToProperties, $ctorArguments, $value);
+            }
+
+            if (!$mappings && $targetRefl->hasProperty($propertyName)) {
+                $value = $this->getSourceValue($source, $mappedTarget, $this->getRawValue($source, $propertyName), $objectMap);
+                $this->storeValue($propertyName, $mapToProperties, $ctorArguments, $value);
+            }
+        }
+
+        if (!$mappingToObject && $ctorArguments) {
+            try {
+                $constructor?->invokeArgs($mappedTarget, $ctorArguments);
```

Comment:
to work with named args? (or using call_user_func_array?)
```suggestion
                $mappedTarget->__construct(...$ctorArguments)
```